src/hookable.ts

```diff
@@ -114,17 +114,14 @@ export class Hookable <
   }
 
   callHook<NameT extends HookNameT> (name: NameT, ...args: Parameters<InferCallback<HooksT, NameT>>) {
-    if (!this._hooks[name]) {
-      return
-    }
-    return serial(this._hooks[name], fn => fn(...args as any))
+    return this.callHookWith(name, hooks => serial(hooks, fn => fn(...args as any)))
   }
 
-  callHookSync<NameT extends HookNameT> (name: NameT, ...args: Parameters<InferCallback<HooksT, NameT>>) {
+  callHookWith<NameT extends HookNameT> (name: NameT, callFn: (fns: HookCallback[]) => any) {
```

Comment:
Maybe also passing `..args`? This can allow to decouple callFn implementation from argument passing.

Reply:
Hm. Do you think that would be better? I'm assuming that as the user already has args, it's cleaner not to pass them around.

```js
const args = []
callHook('name', ...args)
// Current implementation
callHookWith('name', hooks => hooks.map(hook => hook(...args)))
// alternative
callHookWith('name', (hooks, ...args) => hooks.map(hook => hook(...args)), ...args)
```

Reply:
In the current implementation, the user is forced to pass `args` inside the implementation of the handler or create a wrapper.

```js
// Normal calling 
hooks.callhook('name', arg1, arg2)

// Custom handler (current)
const customHandler = hooks => hooks.map(hook => hook(arg1, arg2))
hooks.callHook('name', customHandler)

// Alternative 
const customHandler = (hooks, ...args) => hooks.map(hook => hook(..args))
hooks.callHook('name', customHandler, arg1, arg2)
```

Reply:
implemented what you have in mind, I think, following function signature of your first comment. I take your point about handler definitions, but it still doesn't feel very ergonomic to me. Maybe could feel better if we export a `parallel` util as well? So:

```js
import { parallel } from 'hookable'

hooks.callHook('name', parallel, arg1, arg2)
```

Reply:
> but it still doesn't feel very ergonomic to me.

For argument ordering?

Exposing parallel seems a nice idea BTW